augur/tasks/util/collection_util.py

```diff
@@ -36,6 +36,168 @@ class CollectionState(Enum):
+            AND augur_operations.collection_status.facade_status != '{str(CollectionState.PENDING.value)}'
+            AND augur_operations.collection_status.facade_status != '{str(CollectionState.FAILED_CLONE.value)}'
+            AND augur_operations.collection_status.facade_status != '{str(CollectionState.INITIALIZING.value)}'
+            AND facade_status!='{str(CollectionState.COLLECTING.value)}'
+            AND facade_data_last_collected <= NOW() - INTERVAL '{days_until_collect_again} DAYS'
+        """
+    
+    return condition_concat_string
+
+def get_required_conditions_for_ml_repos(allow_collected_before = False, days_until_collect_again = 1):
+
+    if not allow_collected_before:
+        condition_concat_string = f"""
+            ml_status='{str(CollectionState.PENDING.value)}' AND ml_status!='{str(CollectionState.ERROR.value)}'
+            AND augur_operations.collection_status.secondary_status = '{str(CollectionState.SUCCESS.value)}'
+            AND augur_operations.collection_status.ml_data_last_collected IS NULL
+            AND ml_status!='{str(CollectionState.COLLECTING.value)}'
+        """
+    else:
+        condition_concat_string = f"""
+            ml_status='Success' AND ml_status!='{str(CollectionState.ERROR.value)}'
+            AND augur_operations.collection_status.ml_data_last_collected IS NOT NULL
+            AND ml_status!='{str(CollectionState.COLLECTING.value)}'
+            AND ml_data_last_collected <= NOW() - INTERVAL '{days_until_collect_again} DAYS'
+        """
+    
+    return condition_concat_string
+
+class CollectionRequest:
+    def __init__(self,name,phases,max_repo = 10,new_status = CollectionState.PENDING.value,additional_conditions = None,days_until_collect_again = 1):
+        self.name = name
+        self.phases = phases
+        self.max_repo = max_repo
+        self.days_until_collect_again = days_until_collect_again
+        self.additional_conditions = None
+        self.new_status = new_status
+        self.repo_list = []
+
+        self.status_column = f"{name}_status"
+
+
+        if name == "facade":
+            self.new_status = CollectionState.UPDATE.value
+
+
+    #Get repo urls based on passed in info.
+    def get_valid_repos(self,session):
+        #getattr(CollectionStatus,f"{hook}_status" ) represents the status of the given hook
+        #Get the count of repos that are currently running this collection hook
+        #status_column = f"{hook}_status"
+        active_repo_count = len(session.query(CollectionStatus).filter(getattr(CollectionStatus,f"{self.name}_status" ) == CollectionState.COLLECTING.value).all())
+
+        #Will always disallow errored repos and repos that are already collecting
+
+        #The maximum amount of repos to schedule is affected by the existing repos running tasks
+        limit = self.max_repo-active_repo_count
+
+
+        user_list = split_random_users_list(session,f"{self.name}_status",self.new_status)
+
```

Comment:
I mentioned this below as well, but it appears the splitting of the list into 4 chunks isn't happening in this method call